src/java.naming/share/classes/com/sun/jndi/ldap/sasl/LdapSasl.java

```diff
@@ -130,7 +130,7 @@ public static LdapResult saslBind(LdapClient clnt, Connection conn,
                     cbType = TlsChannelBinding.parseType(
                                 (String)env.get(TlsChannelBinding.CHANNEL_BINDING_TYPE));
                 } catch (ChannelBindingException e) {
-                    throw new SaslException(e.getMessage());
+                    throw new SaslException(e.getMessage(), e);
```

Comment:
Why not ust pass the exception if the API allows? This looks like message duplication.

Reply:
Actually, it turns out I should be throwing `NamingException here`. That is what was being thrown by `TlsChannelBinding.parseType` before and an existing test was expecting that. NamingException only takes a String message. So, there won't be a root cause exception.

Reply:
`NamingException` has `setRootCause()`. Why not use that? I use that one too and full stack is retained.

Reply:
Yes, I can do that. Though it will cause the existing LDAP channel binding test to fail which is checking for an empty root cause. That is checking unspecified behavior and I can change it to check for a `ChannelBindingException` as root cause. So long as we are okay having a non public exception type as the root cause, it's probably helpful to have the full stack there.

Reply:
Yes please retain as much information as possible.